Removes Zend\Expressive\Authentication\ResponsePrototypeTrait

wshafer opened the pull request:
Using PR: https://github.com/zendframework/zend-expressive/pull/561 as a guide this PR removes the ResponsePrototypeTrait in favor of factoryFactories. 

Fixes issue: https://github.com/zendframework/zend-expressive-authentication-oauth2/issues/16

wshafer committed:
Remove dependancy on Zend\Expressive\Authentication\ResponsePrototypeTrait and use factoryFactories instead

wshafer commented:
Ping @webimpress

wshafer committed:
Fix Styles

michalbundyra commented on src/OAuth2Adapter.php:
```diff
@@ -28,20 +28,22 @@ class OAuth2Adapter implements AuthenticationInterface
      * @param ResponseInterface $responsePrototype
      */
-    public function __construct(ResourceServer $resourceServer, ResponseInterface $responsePrototype)
+    public function __construct(ResourceServer $resourceServer, callable $responseFactory)
```
Please remove the whole PHPDoc on `__construct`. It's not updated and we are not using them when there is no any additional information in it.

michalbundyra commented on src/OAuth2Middleware.php:
```diff
@@ -26,20 +26,22 @@ class OAuth2Middleware implements MiddlewareInterface
      *
      * @param AuthorizationServer $server
      * @param ResponseInterface $responsePrototype
      */
```
The same here, please remove PHPDocs

michalbundyra commented on test/OAuth2AdapterFactoryTest.php:
```diff
@@ -10,6 +10,7 @@
 
 namespace ZendTest\Expressive\Authentication\OAuth2;
 
+use function foo\func;
```
Why we need it? Please remove.

michalbundyra commented on test/OAuth2AdapterTest.php:
```diff
@@ -50,7 +54,9 @@ public function testOAuthServerExceptionRaisedDuringAuthenticateResultsInInvalid
-            $this->response->reveal()
+            function () {
+                return $this->response->reveal();
+            }
```
I would add property in that class (and also in other tests), because we have the same usage there many times - and use then just `$this->responseFactory`.

weierophinney committed:
Removes redundant docblocks

weierophinney committed:
Remove unused import

weierophinney committed:
Refactors tests

- Adds property declarations, with annotations detailing types.
- Use a `$responseFactory` property instead of creating a factory each
  time the SUT is instantiated.

weierophinney committed:
Adds CHANGELOG entries for #17

michalbundyra commented on test/OAuth2AdapterFactoryTest.php:
```diff
@@ -52,8 +68,9 @@ public function testInvokeWithResourceServerEmptyResponse()
-            ->willReturn(false);
+            ->get(ResponseInterface::class)
+            ->willReturn(function () {
+            });
```
Also here we can use `this->responseFactory`

weierophinney replied:
No, we cannot, as it is not defined in this class.

In point of fact, this test likely needs to be re-written, as it's supposedly testing that the service can be created if the response service is missing. We should either test that an exception occurs if the service is missing (as `get()` will raise an exception), or that a `TypeError` occurs due to an invalid service. I'll make those changes shortly.

michalbundyra replied:
> No, we cannot, as it is not defined in this class.

Actually it is - line 41: https://github.com/zendframework/zend-expressive-authentication-oauth2/pull/17/files/f9ce613482a2dbf64bb841c7619b3d33f39890b4#diff-d706f562446541d43749de5bbfed80b7R41

michalbundyra commented on test/OAuth2MiddlewareFactoryTest.php:
```diff
@@ -53,8 +53,9 @@ public function testInvokeWithAuthServerWithoutResponseInterface()
-            ->willReturn(false);
+            ->get(ResponseInterface::class)
+            ->willReturn(function () {
+            });
```
And here - `this->responseFactory`

michalbundyra replied:
Ok, here is not defined :)

weierophinney committed:
Refactors factory tests

Ensures that the factories are testing expected behavior:

- TypeError should be raised for a non-callable response factory.
- TypeError should be raised for a response instance returned instead of a response factory.
